### PR TITLE
Set up sentry for reporting errors

### PR DESCRIPTION
**Pull request by matejglejtek**

### CODE REVIEW

**Comment by marianhlavac** [comment]
Rebased nad posledni master

File: lib/main.dart [hunk 1]
```diff
@@ -16,6 +19,40 @@ import 'utils/google_api_key_reader.dart';
...
+const bool kReleaseMode = bool.fromEnvironment('dart.vm.product');
+const bool kProfileMode = bool.fromEnvironment('dart.vm.profile');
+const bool kDebugMode = !kReleaseMode && !kProfileMode;
+
+/// Runs app with Sentry monitoring (only for production environment)
+void runAppWithSentry(void Function() appRunner) async {
+  await runZonedGuarded(() async {
+    if (!kDebugMode) {
+      if (sentryDsn == '') {
+        throw Exception('There is no Sentry DSN specified!');
```
Comment (marianhlavac) [inline lib/main.dart:37]: I think we shouldn't crash the app with an exception when Sentry DSN is missing, it makes development complicated for developers without access to the DSN string, wanting to try the app in the release mode.

File: lib/main.dart [hunk 2]
```diff
@@ -16,6 +19,40 @@ import 'utils/google_api_key_reader.dart';
...
+      }
+      await SentryFlutter.init(
+        (options) {
+          options.dsn = sentryDsn;
+          options.debug = kDebugMode;
```
Comment (marianhlavac) [inline lib/main.dart:42]: Sentry would never run with debug mode enabled, if we prevent calling .init() when kDebugMode==false.

File: lib/main.dart [hunk 3]
```diff
@@ -16,6 +19,40 @@ import 'utils/google_api_key_reader.dart';
 import 'utils/uasid_prefix_reader.dart';
 import 'widgets/app/app.dart';
 
+final sentryDsn = String.fromEnvironment(
```
Comment (marianhlavac) [inline lib/main.dart:22]: For some reason it's not working for me.

Running the app with `❯ flutter run --release --dart-define=SENTRY_DSN=...` doesn't do the trick. Either this should support standard behavior and setting of environment variables, or the setup should be described in README.md or docs/..., but the former is preferred, of course.

**Review by marianhlavac — CHANGES_REQUESTED** [review]
The environment variables didn't work for me, otherwise tested and works.

**Comment by matejglejtek** [comment]
rebased nad posledni master

File: lib/main.dart (hunk 3 above)
Comment (matejglejtek) [reply lib/main.dart]: changed type of sentryDsn from 'final' to 'const', now it works.

File: lib/main.dart (hunk 1 above)
Comment (matejglejtek) [reply lib/main.dart]: fixed, app now continues running

**Review by marianhlavac — APPROVED** [review]
Thanks for fixing the issue with env vars, looks good.

**Pull request merged**